src/server/transaction.cc

```diff
@@ -1266,6 +1266,13 @@ bool Transaction::NotifySuspended(TxId committed_txid, ShardId sid) {
     return false;
   }
 
+  // Wake a transaction only once on the first notify.
+  // We don't care about preserving the strict order with multiple operations running on blocking
+  // keys in parallel, because the internal order is not observable from outside either way.
+  uint64_t default_txid = UINT64_MAX;
+  if (!notify_txid_.compare_exchange_weak(default_txid, committed_txid, memory_order_relaxed))
```

Comment:
compare_exchange_weak is an overkill for "run-once" semantics.
Also, I wonder if `notify_txid_` is needed after this change or it's been used only here

Reply:
Indeed, I forgot to simplify FindFirst.

> compare_exchange_weak is an overkill for "run-once" semantics.

How else should I do run-once? 👀

Reply:
for example,
```
if (var.fetch_add(1, relaxed) == 0) {

}
```